stdlib/public/core/SmallString.swift

```diff
@@ -207,7 +207,7 @@ extension _SmallString {
 
   // Overwrite stored code units, including uninitialized. `f` should return the
   // new count.
-  @inline(__always)
+  @inlinable @inline(__always)
```

Comment:
Could you provide some more motivation here? I really wanted to keep as much of this profusion of machinery out of the ABI and allow the Swift optimizer to boil it away.

This is an ABI change: `withMutableCapacity` was never a symbol in the stdlib and this is adding it. ABI additions should probably be versioned. CC @jrose-apple or @slavapestov is this possible or planned for the near future?

At the very least, have a comment with the Swift version it was added to the ABI. Someone in the future might want to call this method from inlinable code and they should not do so in a back-deployment context.

Reply:
You can put availability on internal functions, sure, even if inlinable. But the calling context has to demonstrate that availability.

Reply:
But can you version inlinability itself?

Reply:
Not yet, but we've got that request on file from y'all. :-)